src/vm/finalizerthread.cpp

```diff
@@ -539,7 +551,17 @@ void FinalizerThread::WaitForFinalizerEvent (CLREvent *event)
                 // Spawn thread to perform the profiler attach, then resume our wait
                 ProfilingAPIAttachDetach::ProcessSignaledAttachEvent();
                 break;
-#endif // FEATURE_PROFAPI_ATTACH_DETACH 
+#endif // FEATURE_PROFAPI_ATTACH_DETACH
+#ifdef FEATURE_PAL
+            case (WAIT_TIMEOUT + kLowMemoryNotification):
```

Comment:
@janvorli The switch above is switch(WaitForMultipleObjectsEx() + uiEventIndexOffsetForWait), so this looks weird but I check for WAIT_TIMEOUT + the two possibilities for uiEventIndexOffsetForWait can be. It avoids having to rewrite the logic the finalizer thread uses to know why it woke up.